### PR TITLE
Feat/addons docs tsup

### PR DESCRIPTION
**Pull request by Saunved**
Issue: https://github.com/storybookjs/storybook/issues/18732

## What I did
Followed the instructions on the issue mentioned above

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots? Yes, with CI
- [ ] Does this need a new example in the kitchen sink apps? No
- [ ] Does this need an update to the documentation? No

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by Saunved** [comment]
@ndelangen 

Based on this deployment failure: https://app.netlify.com/sites/storybookjs/deploys/62def0a92e5b730008aa0ea6
Looks like the docs addon needs to export angular. I can't spot it in the old build, so not sure how it was being accessed.

I am not sure how to export `angular` from this repository with the new build system. Do I add it to exports? Would it be the same for `ember` and `web-components`?

**Comment by Saunved** [comment]
Yes, it seems a bit complicated. I'll push the remaining ones this week for review!

File: code/addons/docs/package.json [hunk 1]
```diff
@@ -46,11 +74,12 @@
   ],
   "scripts": {
     "check": "../../../scripts/node_modules/.bin/tsc --noEmit",
-    "prep": "node ../../../scripts/prepare.js"
+    "prep": "../../../scripts/prepare/bundle.ts"
   },
   "dependencies": {
     "@babel/plugin-transform-react-jsx": "^7.12.12",
     "@babel/preset-env": "^7.12.11",
+    "@digitak/esrun": "^3.2.2",
```
Comment (ndelangen) [inline code/addons/docs/package.json:82]:
```suggestion
```

File: code/yarn.lock [hunk 2]
```diff
@@ -6566,6 +6566,7 @@ __metadata:
     "@babel/core": ^7.12.10
     "@babel/plugin-transform-react-jsx": ^7.12.12
     "@babel/preset-env": ^7.12.11
+    "@digitak/esrun": ^3.2.2
```
Comment (ndelangen) [inline code/yarn.lock:6569]:
```suggestion
```

**Comment by ndelangen** [comment]
Hello @Saunved sorry this PR was somewhat neglected.. it turned out this package was quite complex, and so I took it over at some point for forgot to inform you.

Sorry about that. I hope you don't mind me closing this PR now.

Thank you for your time, energy and effort put into this (and the others) PR. It's greatly appreciated!

**Pull request closed without merging**